fix: add types to package.json

tmsns opened the pull request:
Fixes #2444

For some reason I had to delete [index.d.mts](https://github.com/modernweb-dev/web/pull/2498/files#diff-6b656401880fda83ece317cd724853dba8e6a316b37569a18909f159ddaf307f), to ensure type lookups from other projects was working well. As this file does not exist in `@web/dev-server-core`, I think it should be fine.

tmsns commented:
PR #2468 seems to be similar with some differences:
- doesn't contain the `dev-server-core` package
- doesn't have the removal of the `index.d.mts` file
- but does update other packages as well

What do you guys want to do? @bashmish @koddsson

bashmish commented on packages/dev-server-core/package.json:
```diff
@@ -17,14 +17,17 @@
       "import": "./index.mjs",
-      "require": "./dist/index.js"
+      "require": "./dist/index.js",
+      "types": "./index.d.ts"
```
learned recently from Pascal that https://publint.dev/rules#exports_types_should_be_first

tmsns replied:
Nice! Didn't know. :-)

bashmish commented:
> PR #2468 seems to be similar with some differences:
> 
> * doesn't contain the `dev-server-core` package
> * doesn't have the removal of the `index.d.mts` file
> * but does update other packages as well
> 
> What do you guys want to do? @bashmish @koddsson

I think it would be nice indeed to do this everywhere, as this is public API anyway, so good to export the types as we have them now

I'll review the other PR

bashmish commented:
@tmsns I added dev-server-core to https://github.com/modernweb-dev/web/pull/2468 and fixed the type error which occurred after that addition. I'll merge it once the build succeeds. I assume I can close this PR as the other one solves this for you. Feel free to reopen if smth is still missing.

tmsns commented:
> @tmsns I added dev-server-core to #2468 and fixed the type error which occurred after that addition. I'll merge it once the build succeeds. I assume I can close this PR as the other one solves this for you. Feel free to reopen if smth is still missing.

I saw you only added one of the types of `dev-server-core`.

bashmish commented:
> > @tmsns I added dev-server-core to #2468 and fixed the type error which occurred after that addition. I'll merge it once the build succeeds. I assume I can close this PR as the other one solves this for you. Feel free to reopen if smth is still missing.
> 
> I saw you only added one of the types of `dev-server-core`.

indeed, sorry, fixing it here and merging ;)

bashmish commented on packages/dev-server/package.json:
```diff
@@ -41,9 +41,7 @@
   "files": [
-    "*.d.mts",
     "*.d.ts",
-    "*.js",
```
was publishing a wrong duplicate file of "test-helpers.js", while actually we used the auto-generated mjs one

bashmish commented on packages/dev-server/package.json:
```diff
@@ -41,9 +41,7 @@
     "test:watch": "mocha \"test/**/*.test.mjs\" --watch --watch-files src,test --reporter dot"
   },
   "files": [
-    "*.d.mts",
```
was publishing unused file, which I also removed
